[qa-1571] upgrade sbt to 1.5.5

andy7i opened the pull request:
related to QA-1571 (this pr upgrades sbt, the next one will upgrade the workbenchlibs version)

upgrade sbt version as recommended by cats migration guide (https://typelevel.org/cats-effect/docs/migration-guide)

sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
https://github.com/sbt/sbt/releases/tag/v1.5.0

```
[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```

we need to upgrade many of the rawls dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.

For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.

Side note: Intellij(?) made this really difficult. Changes that I made to build.sbt would not compile correctly until I successfully imported sbt 1.5.5 by running sbt build with only the sbt version change. Additionally, I would make changes to build.sbt (like commenting out the lines I added) and reload the sbt project expecting it to fail, but it would succeed instead. I needed to change other settings (I changed the scalav in Settings.scala) in order for it to pick up the changes in build.sbt.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

andy7i committed:
upgrade sbt version as recommended by cats migration guide (https://typelevel.org/cats-effect/docs/migration-guide)

sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
https://github.com/sbt/sbt/releases/tag/v1.5.0

```
[error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
[error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
[error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
[error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
[error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
[error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
[error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
[error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
[error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
[error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
[error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
[error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
[error]
[error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
[error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
[error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
[error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
[error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
[error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```

we need to upgrade many of these dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.

For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.

Side note: Intellij(?) made this really difficult. Changes that I made to build.sbt would not compile correctly until I successfully imported sbt 1.5.5 by running sbt build with only the sbt version change. Additionally, I would make changes to build.sbt (like commenting out the lines I added) and reload the sbt project expecting it to fail, but it would succeed instead. I needed to change other settings (I changed the scalav in Settings.scala) in order for it to pick up the changes in build.sbt.

Qi77Qi commented:
think it'd be nice to update the image in https://github.com/broadinstitute/rawls/blob/develop/docker/build.sh#L133

andy7i commented:
> think it'd be nice to update the image in https://github.com/broadinstitute/rawls/blob/develop/docker/build.sh#L133

oh good point.. we should probably upgrade this too https://github.com/broadinstitute/firecloud-develop/blob/c7a2abf7bd361fa37d9e1c0c8087e951f8f4cc2b/run-context/local/configs/rawls/docker-rsync-local-rawls.sh.ctmpl#L82

broad's images are pretty out of date so we either need to upgrade and run Broad's images or use something like hseeberger's images (in which case we need to upgrade scalaV as well since there is [no scala 2.12.11 with sbt 1.5.5](https://hub.docker.com/r/hseeberger/scala-sbt/tags?page=1&name=java11_1.5.5_2.12.))

@Qi77Qi you were showing me how leo went with the hseeberger images. Do you know why the team went with that approach, including maintenance costs and security concerns?

Qi77Qi commented:
the broad one is just outdated...we really don't have a reason to main a sbt image ourselves... hseeberger's has been working prefectly fine...it's just an image used in testing, it's not used in prod

s-rubenstein commented on build.sbt:
```diff
@@ -1,6 +1,9 @@
 import Settings._
 import Testing._
 
+ThisBuild / libraryDependencySchemes += "org.typelevel" %% "cats-effect" % "always"
```
Why do we need this here rather than in workbench libs as one of the dependencies there? Alternatively, what does this do? I just notice that there's no other uses of this sort of approach in this file, so am curious!

Qi77Qi replied:
yea these shouldn't be needed

andy7i replied:
the reason for this is in the PR description:

>  sbt 1.5.0 removes eviction warning, and replaces it with stricter eviction error
> https://github.com/sbt/sbt/releases/tag/v1.5.0
> 
> [error] (rawlsCore / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
> [error]
> [error] 	* org.typelevel:cats-core_2.12:2.6.1 (semver-spec) is selected over {2.1.1, 1.6.0, 2.0.0, 2.0.0, 2.2.0}
> [error] 	    +- org.broadinstitute.dsde:rawls-core_2.12:0.1        (depends on 2.6.1)
> [error] 	    +- com.comcast:ip4s-core_2.12:3.0.3                   (depends on 2.6.1)
> [error] 	    +- org.typelevel:cats-effect-kernel_2.12:3.2.9        (depends on 2.6.1)
> [error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.6.1)
> [error] 	    +- org.typelevel:cats-mtl_2.12:1.2.1                  (depends on 2.6.1)
> [error] 	    +- org.typelevel:log4cats-core_2.12:2.1.1             (depends on 2.6.1)
> [error] 	    +- co.fs2:fs2-core_2.12:3.1.1                         (depends on 2.6.1)
> [error] 	    +- io.circe:circe-core_2.12:0.14.1                    (depends on 2.6.1)
> [error] 	    +- org.broadinstitute.dsde:workbench-util_2.12:0.1-1fd7bae7b-SNAP (depends on 2.6.1)
> [error] 	    +- io.lemonlabs:scala-uri_2.12:3.0.0                  (depends on 2.2.0)
> [error] 	    +- io.chrisdavenport:vault_2.12:2.0.0                 (depends on 2.0.0)
> [error] 	    +- io.chrisdavenport:unique_2.12:2.0.0                (depends on 2.0.0)
> [error] 	    +- io.chrisdavenport:log4cats-slf4j_2.12:0.3.0        (depends on 1.6.0)
> [error] 	    +- com.github.sebruck:opencensus-scala-core_2.12:0.7.2 (depends on 2.1.1)
> [error]
> [error] 	* org.typelevel:log4cats-slf4j_2.12:2.1.1 (early-semver) is selected over 1.2.2
> [error] 	    +- org.broadinstitute.dsde.workbench:workbench-util2_2.12:0.2-ae11b9f (depends on 2.1.1)
> [error] 	    +- org.broadinstitute.dsde.workbench:workbench-google2_2.12:0.21-9d25534 (depends on 1.2.2)
> [error]
> [error] 	* co.fs2:fs2-core_2.12:3.1.1 (early-semver) is selected over {3.0.4, 2.5.6, 2.2.2, 3.0.4, 2.5.6, 2.2.2}
> [error] 	    +- co.fs2:fs2-io_2.12:3.1.1                           (depends on 3.1.1)
> [error] 	    +- org.http4s:jawn-fs2_2.12:1.0.0                     (depends on 2.2.2)
> [error] 	    +- org.http4s:http4s-core_2.12:0.21.23                (depends on 2.5.6)
> [error] 	    +- io.circe:circe-fs2_2.12:0.14.0                     (depends on 3.0.4)
> [error]
> [error]
> [error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
> we need to upgrade many of the rawls dependencies, and we may want to do that across the board here. (I've added this to our tech debt notes doc https://docs.google.com/document/d/1r0JWqBvmT9Y6JjT8sUrHTzNIE0Ed8gQt8iLhs0DmAbA/edit) The Leonardo repo keeps all of their dependencies up to date using a bot as well as using one version hash for all workbench-libs: https://github.com/DataBiosphere/leonardo/blob/develop/project/Dependencies.scala#L19. We may want to do something similar.
> 
> For now, there are 2 ways to opt out from this error (given by the sbt team: https://eed3si9n.com/sbt-1.5.0#:~:text=slash%20for%20details.-,Eviction%20error,-sbt%201.5.0%20removes) which is either to set all eviction errors back to INFO logs, or to specify which modules to ignore. I went with the latter option since it is more specific.
> 
>

gpolumbo-broad reviewed (approved):
Looks fine to me.  Thanks @Qi77Qi for helping with this

s-rubenstein reviewed (approved):
LGTM

andy7i committed:
update scala, sbt versions

we need to upgrade scalaV as well since there is no scala 2.12.11 with sbt 1.5.5)

andy7i committed:
one more

andy7i committed:
try 2.12.14

andy7i committed:
sbt-scoverage upgrade

fixes errors saying:
```
java.lang.NoSuchMethodError: scala/tools/nsc/Global.reporter()Lscala/tools/nsc/reporters/Reporter
```

andy7i committed:
Revert "try 2.12.14"

This reverts commit 44d0fce34ba6a6a68b8f2bc64e7c1b313e6cd468.

andy7i committed:
// For the following errors:
```
    //[error] java.lang.RuntimeException: deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn$.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn$.class
    //[error] deduplicate: different file contents found in the following:
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar:scala/annotation/nowarn.class
    //[error] /home/sbtuser/.cache/coursier/v1/https/repo1.maven.org/maven2/org/http4s/http4s-core_2.12/0.21.23/http4s-core_2.12-0.21.23.jar:scala/annotation/nowarn.class
```

andy7i committed:
Edmund ran sbt assembly without each line to see which ones were still used. This removes the lines that are no longer used.

andy7i committed:
need hash even for publishing rawls model

this handles the hash without needing to install git on the docker image (which, fun fact, does not have apt, yum, dpkg, or even sudo)

 ```
 14:42:55 [info] resolving key references (11031 settings) ...
 14:42:55 java.io.IOException: Cannot run program "git": error=2, No such file or directory
 14:42:55 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)

```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4805/console

andy7i committed:
get the git hash for the repo, not just the model

andy7i commented on docker/build.sh:
```diff
@@ -100,15 +100,15 @@ function make_jar()
 
-    # Get the last commit hash of the model directory and set it as an environment variable
-    GIT_MODEL_HASH=$(git log -n 1 --pretty=format:%h model)
+    # Get the last commit hash and set it as an environment variable
+    GIT_HASH=$(git log -n 1 --pretty=format:%h)
```
not sure if there's any benefit to getting the hash from the model directory. I could be wrong about this though.

marctalbott replied:
I would lean towards leaving it alone if it's working fine unless there's a particular reason you need to make this change

andy7i replied:
changed it back

andy7i replied:
I'm seeing 403s during the rawls-build job, when it publishes rawls-model with the `model` folder/project hash. This is happening on each commit.

I looked at this unrelated PR https://github.com/broadinstitute/rawls/pull/1555/files which does not touch `model` and it publishes a new version of the rawls-model to jfrog on every commit.

Since this is de facto the behavior, I'm now using GIT_HASH (and removed references to `model`). Whether this is the right thing to do, we could create a tech debt ticket to handle this. I'm not sure if it's worth tackling this so I haven't created the ticket but can create one if you (or anyone else) feels like we should spend time on this.

andy7i commented on project/Version.scala:
```diff
@@ -6,13 +6,12 @@ object Version {
   val baseModelVersion = "0.1"
 
   def getVersionString = {
-    def getLastModelCommitFromGit = { s"""git log -n 1 --pretty=format:%h""" !! }
```
need hash even for publishing rawls model. this handles the hash without needing to install git on the docker image (which, fun fact, does not have apt, yum, dpkg, or even sudo)

 ```
 14:42:55 [info] resolving key references (11031 settings) ...
 14:42:55 java.io.IOException: Cannot run program "git": error=2, No such file or directory
 14:42:55 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)

```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4805/console

marctalbott replied:
Out of curiosity, why the change from `def` to `lazy val`?

andy7i replied:
I was seeing errors where Rawls was trying to run a git command and failing. I figured that the hash only needed to be set once and wouldn't change. Since a `lazy val` is evaluated only when it is used, I thought this could work as well. It seems `def`s are also only evaluated when called so either seems fine. I changed it back just in case.

andy7i commented on project/Merging.scala:
```diff
@@ -2,23 +2,16 @@ import sbtassembly.{MergeStrategy, PathList}
 
 object Merging {
   def customMergeStrategy(oldStrategy: (String) => MergeStrategy):(String => MergeStrategy) = {
-    case x if x.endsWith("io.netty.versions.properties") => MergeStrategy.discard
```
Edmund ran sbt assembly without these lines and there were no errors.

marctalbott replied:
What are these `MergeStrategy`s? Are we removing them because we think they're unnecessary or because they're incompatible?

Qi77Qi replied:
if `sbt assembly` doesn't complain, it's probably fine...it's possible some of these rules were added when there were conflicts between libraries, but after libraries are upgraded, they don't apply anymore

andy7i commented on project/plugins.sbt:
```diff
@@ -2,10 +2,10 @@ addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.7")
 addSbtPlugin("io.spray" % "sbt-revolver" % "0.9.1")
 
-addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.6.1")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.9.1")
```
sbt-scoverage upgrade fixes errors saying:
```
java.lang.NoSuchMethodError: scala/tools/nsc/Global.reporter()Lscala/tools/nsc/reporters/Reporter
```

s-rubenstein reviewed (approved):
LGTM

andy7i committed:
PR feedback - rename GIT_HASH back to GIT_MODEL_HASH

andy7i commented:
jenkins retest
build failed with:
```
10:47:29 [error] (rawlsModel / publish) java.io.IOException: Server returned HTTP response code: 403 for URL: https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634827618/org/broadinstitute/dsde/rawls-model_2.12/0.1-efac163-SNAP/rawls-model_2.12-0.1-efac163-SNAP.pom
```

andy7i commented:
jenkins retest -- same error.. I've seen this error before (https://github.com/broadinstitute/rawls/pull/1507#issuecomment-911695286) and it may be a flaky issue.

andy7i commented:
jenkins retest - doesn't seem like a flaky issue. rerunning while working on other dependency updates for qa-1571

andy7i committed:
dummy commit

andy7i commented:
jenkins retest - 15:14:52 POST https://us-central1-broad-workbench-tos-qa.cloudfunctions.net/tos/v1/user/response: 400

andy7i committed:
Merge branch 'develop' into aj-sr--upgrade-sbt-version--qa-1571

andy7i commented:
jenkins retest -- concerned about this test failure:
```
Error Message
Futures timed out after [5 minutes]
Stacktrace
      java.util.concurrent.TimeoutException: Futures timed out after [5 minutes]
      at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:259)
      at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:263)
      at scala.concurrent.Await$.$anonfun$result$1(package.scala:223)
      at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:57)
      at scala.concurrent.Await$.result(package.scala:146)
      at org.broadinstitute.dsde.workbench.service.RestClient.sendRequest(RestClient.scala:63)
      at org.broadinstitute.dsde.workbench.service.RestClient.requestWithJsonContent(RestClient.scala:107)
      at org.broadinstitute.dsde.workbench.service.RestClient.postRequest(RestClient.scala:132)
      at org.broadinstitute.dsde.workbench.service.RestClient.postRequest$(RestClient.scala:129)
      at org.broadinstitute.dsde.workbench.service.Orchestration$.postRequest(Orchestration.scala:855)
      at org.broadinstitute.dsde.workbench.service.Orchestration$methods$.createMethod(Orchestration.scala:551)
      at org.broadinstitute.dsde.workbench.fixture.MethodFixtures.$anonfun$withMethod$1(MethodFixtures.scala:17)
      at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
      at org.broadinstitute.dsde.workbench.fixture.MethodFixtures.withMethod(MethodFixtures.scala:16)
      at org.broadinstitute.dsde.workbench.fixture.MethodFixtures.withMethod$(MethodFixtures.scala:11)
      at org.broadinstitute.dsde.test.api.WorkspaceApiSpec.withMethod(WorkspaceApiSpec.scala:36)
      at org.broadinstitute.dsde.test.api.WorkspaceApiSpec.$anonfun$new$74(WorkspaceApiSpec.scala:456)
      at org.broadinstitute.dsde.test.api.WorkspaceApiSpec.$anonfun$new$74$adapted(WorkspaceApiSpec.scala:439)
      at org.broadinstitute.dsde.workbench.fixture.WorkspaceFixtures.$anonfun$runTestAndCleanUpWorkspace$1(WorkspaceFixtures.scala:26)
```
https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-fiab-test-runner/37924/testReport/junit/org.broadinstitute.dsde.test.api/WorkspaceApiSpec/Rawls_should_not_allow_readers_to_import_method_configs_from_another_workspace/

andy7i committed:
Revert "PR feedback - rename GIT_HASH back to GIT_MODEL_HASH"

This reverts commit 9f78b28b68b2d7a96d65da0a4f1e716c5d15e7a8.

andy7i committed:
change back to def. ; this fixes 403s from jfrog when there are no changes to model but we still need to publish new changes.
not 100% sure how this worked before but without spending time to investigate my guess is that it was actually using the repo's git hash (and NOT the model git hash).

example:
first PR:
```

11:58:01 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.pom
11:58:01 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.jar
11:58:02 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP-sources.jar
11:58:02 [info] 	published rawls-model_2.12 to https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918248/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP-javadoc.jar
```
from https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4850/console

subsequent commits all had this error:
```
12:00:54 [error] java.io.IOException: Server returned HTTP response code: 403 for URL: https://broadinstitute.jfrog.io/broadinstitute/libs-release-local;build.timestamp=1634918418/org/broadinstitute/dsde/rawls-model_2.12/0.1-c3e37ec-SNAP/rawls-model_2.12-0.1-c3e37ec-SNAP.pom
12:00:54 [error] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)

```
https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-build/4851/console

andy7i committed:
Revert "dummy commit"

This reverts commit 03fce34843b36e1275543eb865904871618e9fb0.
